src/countBy.ts

```diff
@@ -0,0 +1,57 @@
+import { purry } from "./purry";
+
+/**
+ * Categorize and count elements in an array using a defined callback function. The transformation function applies to each element in the array to determine its category and then counts how many elements fall into each category.
+ *
+ * @param data - The array.
+ * @param fn - The transformation function.
+ * @signature
+ *   R.countBy(array, fn)
+ * @example
+ *    R.countBy(
+ *      ["a", "b", "c", "B", "A", "a"],
+ *      (x) => x.toLowerCase()
```

Comment:
We provide a toLowerCase utility, let's use it in the example too.

```suggestion
 *      R.toLowerCase()
```